pkg/controlplane/controller/leaderelection/leaderelection_controller.go

```diff
@@ -402,7 +402,11 @@ func (c *Controller) reconcileElectionStep(ctx context.Context, leaseNN types.Na
 	}
 
 	if reflect.DeepEqual(existing, orig) {
-		klog.V(5).Infof("Lease %s already has the most optimal leader %q", leaseNN, *existing.Spec.HolderIdentity)
+		leader := ""
+		if existing.Spec.HolderIdentity != nil {
+			leader = *existing.Spec.HolderIdentity
+		}
+		klog.V(5).Infof("Lease %s already has the most optimal leader %q", leaseNN, leader)
```

Comment:
This comment feels a bit weird if we're using third party strategy since leader may be nil. Maybe we should log something different like 
`"Lease %s is managed by a third party strategy %s"`